src/cursor.c

```diff
@@ -882,6 +882,47 @@ cursor_button(struct wl_listener *listener, void *data)
 	}
 }
 
+bool
+handle_cursor_axis(struct server *server, struct cursor_context *ctx,
+		struct wlr_pointer_axis_event *event)
+{
+	struct mousebind *mousebind;
+	bool activated_any = false;
+
+	uint32_t modifiers = wlr_keyboard_get_modifiers(
+			&server->seat.keyboard_group->keyboard);
+
+	uint32_t button = 0;
+	if (event->delta_discrete < 0) {
+		button = BTN_GEAR_UP;
+	} else if (event->delta_discrete > 0) {
+		button = BTN_GEAR_DOWN;
+	}
+
+	if (!button || event->source != WLR_AXIS_SOURCE_WHEEL
+			|| event->orientation != WLR_AXIS_ORIENTATION_VERTICAL) {
+		wlr_log(WLR_DEBUG, "Failed to handle cursor axis event");
+		return false;
+	}
+
+	wl_list_for_each(mousebind, &rc.mousebinds, link) {
+		if (ssd_part_contains(mousebind->context, ctx->type)
+				&& mousebind->button == button
+				&& modifiers == mousebind->modifiers) {
+			switch (mousebind->mouse_event) {
+			case MOUSE_ACTION_CLICK:
+				activated_any = true;
+				actions_run(ctx->view, server, &mousebind->actions,
+					/*resize_edges*/ 0);
```

Comment:
If we only want to support the `Up` / `Down` + `Click` combination we could also replace the switch with a simple
```c
if (mousebind->mouse_event == MOUSE_ACTION_CLICK) {
	actions_run(..)
}
```